matsim/src/main/java/org/matsim/core/mobsim/hermes/HermesBuilder.java

```
@@ -23,36 +23,55 @@
 import org.matsim.core.mobsim.qsim.AbstractQSimModule;
+import org.matsim.core.mobsim.qsim.components.QSimComponentsConfig;
 import org.matsim.core.mobsim.qsim.components.QSimComponentsConfigurator;
+import org.matsim.core.router.util.TravelTime;
 
-class HermesBuilder implements AllowsConfiguration {
+import java.util.*;
+
+public class HermesBuilder implements AllowsConfiguration {
+	private final Collection<AbstractQSimModule> qsimModules = new LinkedList<>();
+	private final QSimComponentsConfig components = new QSimComponentsConfig();
+
+	private final List<AbstractModule> overridingControllerModules = new LinkedList<>();
+	private final List<AbstractQSimModule> overridingQSimModules = new LinkedList<>();
+	private final Map<String, TravelTime> travelTimes = new HashMap<String, TravelTime>();
 
 	@Override
-	public AllowsConfiguration addOverridingModule(AbstractModule abstractModule) {
-		// TODO Auto-generated method stub
-		return null;
+	public HermesBuilder addOverridingModule(AbstractModule abstractModule) {
+		overridingControllerModules.add(abstractModule);
+		return this;
 	}
 
 	@Override
-	public AllowsConfiguration addOverridingQSimModule(AbstractQSimModule qsimModule) {
-		// TODO Auto-generated method stub
-		return null;
+	public HermesBuilder addOverridingQSimModule(AbstractQSimModule qsimModule) {
+		this.overridingQSimModules.add(qsimModule);
+		return this;
 	}
 
 	@Override
-	public AllowsConfiguration addQSimModule(AbstractQSimModule qsimModule) {
-		// TODO Auto-generated method stub
-		return null;
+	public HermesBuilder addQSimModule(AbstractQSimModule qsimModule) {
+		this.qsimModules.add(qsimModule);
+		return this;
 	}
 
 	@Override
-	public AllowsConfiguration configureQSimComponents(QSimComponentsConfigurator configurator) {
-		// TODO Auto-generated method stub
-		return null;
+	public HermesBuilder configureQSimComponents(QSimComponentsConfigurator configurator) {
+		configurator.configure(components);
+		return this;
 	}
```

Not sure this is required

---

How do you mean? I guess those QSim methods had empty implementations thus far because nobody would be trying to add QSim config to Hermes? 

It's maybe a design smell that the `AllowsConfiguration` interface has a bunch of QSim-specific methods in there anyway.

If it's not appropriate or valid to pass QSim modules and config into the Hermes builder, then those methods should perhaps throw an [`UnsupportedOperationException`](UnsupportedOperationException). On the other hand, if it's a perfectly valid thing to do, then your changes are good. Either way, it's kind of bad form that previously this builder exposed those methods, but they were silently no-ops.